### PR TITLE
Refine IndexHandler methods a bit to make them reentrant

### PR DESCRIPTION
**Pull request by klsince**
Use a defensive copy to make the IndexHandler methods reentrant and less confusing to use. This is not an issue today, as IndexHandler object is not reused.

### CODE REVIEW

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/FSTIndexHandler.java [hunk 1]
```diff
@@ -74,22 +74,23 @@ public class FSTIndexHandler implements IndexHandler {
   public FSTIndexHandler(SegmentMetadata segmentMetadata, IndexLoadingConfig indexLoadingConfig) {
     _segmentMetadata = segmentMetadata;
     _fstType = indexLoadingConfig.getFSTIndexType();
-    _columnsToAddIdx = new HashSet<>(indexLoadingConfig.getFSTIndexColumns());
+    _columnsToAddIdx = indexLoadingConfig.getFSTIndexColumns();
```
Comment (vvivekiyer) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/FSTIndexHandler.java:77]: (nit) Isn't it probably cleaner and safer (from future changes) to not have `_columnsToAddIdx` as a private member?  Instead just have `_indexLoadingConfig` as a private member and compute `columnsToAddIdx` in the respective methods?

This is the pattern followed in other handlers like BloomFilterHandler,H3IndexHandler.

**Review by vvivekiyer — APPROVED** [review]
minor comment. LGTM.

**Comment by somandal** [comment]
lgtm overall

**Review by somandal — APPROVED** [review]

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/FSTIndexHandler.java (hunk 1 above)
Comment (klsince) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/FSTIndexHandler.java:77]: I see BF/H3 hander doesn't hold indexLoadingConfig reference either, but configs for creating BF/H3. 
The handlers in this PR don't have specific configs from indexLoadingConfig either, so not hold its reference.

**Review by jackjlli — APPROVED** [review]
LGTM

**Pull request merged**